src/commands/connect/plugin.ts

```diff
@@ -298,6 +306,21 @@ const connectComponentConfig = async (
     );
 };
 
+const getMetadata = (plugins: ConnectPluginInstance[] = []): undefined | Record<string, unknown> => (
+    plugins.reduce<undefined | Record<string, unknown>>(
+        (acc, plugin) => {
+            if (!plugin.metadata) {
+                return acc;
+            }
+            if (!acc) {
+                return acc;
```

Comment:
```suggestion
                return { ...plugin.metadata };
```
Otherwise it would always return undefined as the initialValue is undefined

Reply:
I think instead of `undefined` metadata should at least be an empty object